### PR TITLE
sync blocker for auto toggle

### PR DESCRIPTION
**Pull request by veebuv**

### CODE REVIEW

**Comment by veebuv** [comment]
The value sent back on changed will still be the toggled value so the emit event can send that to the parent, the parent can act accordingly, make API calls etc, and once data is resolved, set the value to true, which will auto toggle

**Comment by ninhxuanluu** [comment]
need to review asap @euvl, pls

**Comment by Sekonda** [comment]
Can this be merged @euvl? Thanks

File: src/Button.vue [hunk 1]
```diff
@@ -269,14 +272,18 @@ export default {
     }
   },
   methods: {
-    toggle (event) {
-      this.toggled = !this.toggled
-      this.$emit('input', this.toggled)
+    toggle(event) {
+      const toggled = !this.toggled;
+      if (!this.sync) {
+        this.toggled = toggled;
```
Comment (euvl) [inline src/Button.vue]:
```suggestion
        // TODO: Unit test it
        this.toggled = toggled;
```

**Pull request merged**